web/src/poll_widget.ts

```diff
@@ -225,10 +228,9 @@ export function activate({
 
     function render_results(): void {
         const widget_data = poll_data.get_widget_data();
-
         const html = render_widgets_poll_widget_results(widget_data);
         $elem.find("ul.poll-widget").html(html);
-
+        render_poll_timestamp($elem);
```

Comment:
Directly pass `$elem.find("ul.poll-widget")` instead to this function. That will make it more generalised, so also reusable by todo widgets, eliminating the near duplicate function in `todo_widget.js`

Reply:
Would be even better if you could extract this as a function in `rendered_markdown` where this code is from, and call that here. 

We always want to minimize code duplication as much as possible

Reply:
I felt using the `update_elements` of `rendered_markdown` would be unreasonable since that method would try to look out for lot of other elements apart form `time` in `$elem` like mentions, codehilite,... but those for sure won't be rendered inside the widget by the inline processor.

Moreover, importing `rendered_markdown.js` method into `poll_widget.ts` is causing `Could not find a declaration file for module './rendered_markdown'. 'rendered_markdown.js' implicitly has an 'any' type`

Reply:
I meant extracting just the relevant code as a new function and calling that, not the pre existing whole `update_elements` function.

But yes, the non typescript file import into this typescript file still remains an issue :thinking: